src/api/movieService.ts

```diff
@@ -1,43 +1,114 @@
-import { movieList } from "../assets/data/constData";
-import { Movie } from "../models/movie";
-
-let movies = [...movieList];
-
-export const fetchMovies: () => Promise<Movie[]> = () => {
-  return new Promise((res, rej) => {
-    return setTimeout(() => res(movies), 0);
-  });
-};
-
-export const addMovie = (newMovie: Movie) => {
-  return new Promise((res, rej) =>
-    setTimeout(() => {
-      movies = [...movies, newMovie];
-      return res(true);
-    }, 1000)
-  );
-};
-
-export const editMovie = (movie: Movie, id: string) => {
-  return new Promise((res, rej) =>
-    setTimeout(() => {
-      let updatedList = [...movies];
-      const indexOfMovie = updatedList.findIndex((movie) => movie.id === id);
-      updatedList[indexOfMovie] = movie;
-      movies = updatedList;
-      return res(true);
-    }, 1000)
-  );
-};
-
-export const deleteMovie = (id: string | undefined) => {
-  return new Promise((res, rej) =>
-    setTimeout(() => {
-      let updatedList = [...movies];
-      const indexOfMovie = updatedList.findIndex((movie) => movie.id === id);
-      updatedList.splice(indexOfMovie, 1);
-      movies = updatedList;
-      return res(true);
-    }, 1000)
-  );
-};
+const baseUrl = 'http://localhost:4000/movies';
+const baseFetch = async (url: string, params?: any) => {
+  try {
+    const response = await fetch(url, params);
+    if (!response.ok) {
+      throw new Error("HTTP status code: " + response.status)
+    }
+
+    if (response.statusText === 'No Content' && response.ok){
+      return true;
+    }
+    const data = await response.json();
+    return data;
+  } catch (e) {
+      throw new Error('Something went wrong');
+  }
+}
+
+export const get = async (url: string, params?: any) => {
+  return baseFetch(url, {
+    ...params,
+    method: 'get'
+  })
+}
+
+export const post = async (url: string, body: any, params?: any) => {
+  return baseFetch(url, {
+    ...params,
+    ...body,
+    method: 'post',
+    headers: {
+      ...params?.headers,
+      'Content-Type': 'application/json;charset=utf-8'
+    },
+  })
+}
+
+export const put = async (url: string, body: any, params?: any) => {
+  return baseFetch(url, {
+    ...params,
+    ...body,
+    method: 'put',
+    headers: {
+      ...params?.headers,
+      'Content-Type': 'application/json;charset=utf-8'
+    },
+  })
+}
+
+export const deleteRequest = async (url: string, params?: any) => {
+  return baseFetch(url, {
+    ...params,
+    method: 'delete',
+    headers: {
+      ...params?.headers,
+      'Content-Type': 'application/json;charset=utf-8'
+    },
+  })
+}
+
+export const getMovies = (queries: { sortBy: string, filter: string }) => {
+  const url = getUrlWithQuery(queries);
+  return get(url);
+}
+
+const getUrlWithQuery = (queries: { sortBy: string, filter: string }) => {
+  let urlString = Object.entries(queries).map((el, index) =>{
```

Comment:
```suggestion
  let urlString = Object.entries(queries).map(([key, value], index) =>{
```
This will help you avoid using "magic" numbers for `el[0]` and `el[1]`.